### PR TITLE
Discord event syncing

### PR DESCRIPTION
**Pull request by calidbaba**
- Events made on the website is also created as an discord event on the discord server
- Updating and deleting is also synced
- Most of the logic is in src/server/utils.js, maybe it should be in the src/server/discord.js. However that seemed to be more logic connected with discord login.

### CODE REVIEW

File: src/routes/arrangementer.svelte [hunk 1]
```diff
@@ -51,8 +52,30 @@
 	const refresh = async () => {
 		events = (await api.getEvents(showOld)).json;
 	};
-
+	const skjulArrangemant = () => {
```
Comment (kristianvld) [inline src/routes/arrangementer.svelte:55]: Please keep it to english in the chat folks 🙃

File: src/routes/arrangementer.svelte [hunk 2]
```diff
@@ -51,8 +52,30 @@
...
+		//you cant share a hidden event to discord
+		//also tried to disable the other checkboxs if one is selected, but it was hard to see with the colorscheme and made things confusing
+		if (newEvent.discord) {
+			error = 'Du kan ikke både skjule et arrangement og dele det på discord';
+			newEvent.discord = false;
+			return;
+		}
+	};
+	const delArrangemangtDiscord = () => {
```
Comment (kristianvld) [inline src/routes/arrangementer.svelte:64]: English please

File: src/routes/arrangementer.svelte [hunk 3]
```diff
@@ -113,7 +136,10 @@
 				<label><span class="col">CTF Navn:</span> <input type="text" bind:value={newEvent.ctf.name} /></label>
 				<label><span class="col">CTF Link:</span> <input type="text" bind:value={newEvent.ctf.url} /></label>
 				<label><span class="col">Info:</span> <textarea rows="3" bind:value={newEvent.info} /></label>
-				<label><span class="col">Skjult:</span> <input type="checkbox" bind:value={newEvent.hidden} bind:checked={newEvent.hidden} /></label>
+				<label><span class="col">Skjult:</span> <input type="checkbox" bind:value={newEvent.hidden} bind:checked={newEvent.hidden} on:change={skjulArrangemant} /></label>
```
Comment (kristianvld) [inline src/routes/arrangementer.svelte:139]: The "value" attributes of checkboxes has no function or effect. Use `checked` only.

File: src/server/utils.js [hunk 4]
```diff
@@ -24,7 +27,80 @@ export const removeEmptyObjects = (obj, cache) => {
 		}
 	}
 };
+//deletes discord event with id = eventId
+export const deleteDiscordEvent = async (eventId) => {
```
Comment (kristianvld) [inline src/server/utils.js:31]: Please move this to discord.js

File: src/server/utils.js [hunk 5]
```diff
@@ -24,7 +27,80 @@ export const removeEmptyObjects = (obj, cache) => {
...
+	return await fetchResource(`https://discord.com/api/v8/guilds/${process.env.DISCORD_SERVER_ID}/scheduled-events/${eventId}`, {
```
Comment (kristianvld) [inline src/server/utils.js:32]: Please use the discordFetch function, no need for duplicate code doing the same thing.

File: src/server/utils.js [hunk 6]
```diff
@@ -24,7 +27,80 @@ export const removeEmptyObjects = (obj, cache) => {
...
+		errorText: 'ERROR',
+		fetch: fetch,
+		headers: {
+			Authorization: `Bot ${process.env.DISCORD_BOT_TOKEN}`
+		}
+	});
+};
+//makes a discord event when an event is made on the website, takes in form data from the make-event form.
+//if isUpdate == true it updates an existing event instead
+export const makeDiscordEvent = async (jsonData, isUpdate) => {
```
Comment (kristianvld) [inline src/server/utils.js:43]: Please move this to discord.js

**Review by kristianvld — CHANGES_REQUESTED** [review]

File: src/routes/arrangementer.svelte (hunk 2 above)
Comment (calidbaba) [reply src/routes/arrangementer.svelte]: shieet

**Review by kristianvld — APPROVED** [review]

**Pull request merged**